src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpNavigateTo.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Common;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpNavigateTo: AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpNavigateTo(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpNavigateTo))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.SignatureHelp)]
+        public void NavigateTo1()
+        {
+            AddFile("test1.cs", open:false, contents: @"
+class FirstClass
+{
+    void FirstMethod() { }
+}");
+
+
+            AddFile("test2.cs", open: true, contents: @"
+");
+
+            WaitForAsyncOperations(FeatureAttribute.Workspace);
```

Comment:
Please no waiting on workspaces directly. Please move it to the helper that needs the wait, and justify it's need to be there.